### DISCUSSION TITLE
Neo-tree performance problems

### DISCUSSION THREAD

**Discussion by Old-Farmer** [LazyVim/LazyVim#3017] [General]
Neo-tree is good, but when I work on a big codebase, it seems too slow:

1. opening the sidebar always takes a lot of time, maybe 1 or 2 seconds.
2. following the current file is also very slow, maybe 1 or 2 seconds.

Any suggestion for using this plugin, or should I switch to another plugin or just use netrw.

**Comment by 0x2E** [comment]
I tried in Linux source code (big enough?), no performance issue.

My conf:

```lua
  {
    "nvim-neo-tree/neo-tree.nvim",
    opts = {
      window = {
        position = "right",
      },
      filesystem = {
        filtered_items = {
          hide_dotfiles = false,
          hide_gitignored = true,
        },
      },
    },
  },
```

**Comment by awilkins** [comment]
Agree with the OP ; updating the sidebar in a codebase which has 4,700 files in it creates a noticeable pause (e.g. open a split with two files from this folder, and move back and forth between them).

I decided that having the tree open was pointless when opening fzf was so fast ...

**Comment by prerit714** [comment]
Try using oil.nvim, I find to be a really good alternative for it.
